os/kernel/sched/Make.defs

```diff
@@ -60,7 +60,7 @@ CSRCS += sched_setscheduler.c sched_getscheduler.c
 CSRCS += sched_yield.c sched_rrgetinterval.c sched_foreach.c
 CSRCS += sched_lock.c sched_unlock.c sched_lockcount.c sched_self.c
 CSRCS += sched_getaffinity.c sched_setaffinity.c
-CSRCS += sched_getcpu.c
+CSRCS += sched_getcpu.c sched_checkstackoverflow.c
```

Comment:
The function is called with CONFIG_SW_STACK_OVERFLOW_DETECTION. 
It means this file is valid with the config. Let's add conditional for this.
Or, remove the conditional at all c file, and in header, let's do like below
```
#ifdef CONFIG_XXX
void sched_checkstackoverflow(FAR struct tcb_s *rtcb);
#else
#define sched_checkstackoverflow(x)
#endif
```